src/components/TDViewer/TDViewer.jsx

```diff
@@ -98,15 +127,141 @@ export default function TDViewer() {
 
-    const alreadyRenderedKeys = ["id", "properties", "actions", "events", "description", "title",];
+    const alreadyRenderedKeys = ["id", "properties", "actions", "events", "forms", "description", "title",];
 
-    const attributeListObject = buildAttributeListObject(tdJSON.id ? {id: tdJSON.id} : {}, tdJSON, alreadyRenderedKeys);
+    const attributeListObject = buildAttributeListObject(tdJSON.id ? { id: tdJSON.id } : {}, tdJSON, alreadyRenderedKeys);
 
     const attributes = Object.keys(attributeListObject).map(x => {
         return <li key={x}>{x} : {JSON.stringify(attributeListObject[x])}</li>
     });
 
 
+    const sortKeysInObject = (kind) => {
+        const ordered = {};
+        const toSort = Object.keys(tdJSON[kind]).map(x => {
+            return { key: x, title: tdJSON[kind][x].title }
+        })
+        if (sortorder === 'asc') {
+            toSort.sort(function (a, b) {
+                var nameA = a.title ? a.title.toUpperCase() : a.key.toUpperCase();
+                var nameB = b.title ? b.title.toUpperCase() : b.key.toUpperCase();
+                if (nameA < nameB) {
+                    return -1;
+                }
+                if (nameA > nameB) {
+                    return 1;
+                }
+                return 0;
+            }).forEach(function (sortedObject) {
+                ordered[sortedObject.key] = tdJSON[kind][sortedObject.key];
+            });
+            sortorder = 'desc'
+        } else {
+            toSort.sort(function (a, b) {
+                var nameA = a.title ? a.title.toUpperCase() : a.key.toUpperCase();
+                var nameB = b.title ? b.title.toUpperCase() : b.key.toUpperCase();
+                if (nameA < nameB) {
+                    return -1;
+                }
+                if (nameA > nameB) {
+                    return 1;
+                }
+                return 0;
+            }).reverse().forEach(function (sortedObject) {
+                ordered[sortedObject.key] = tdJSON[kind][sortedObject.key];
+            });
+            sortorder = 'asc'
+        }
+        tdJSON[kind] = ordered
+        context.updateOfflineTD(JSON.stringify(tdJSON, null, 2))
+    }
+
+    const search = (event) => {
+        if (first) {
+            console.log(first)
+            unfilteredProps = tdJSON.properties
+            first = false
+        }
+        let sortedProps = {};
+        if (event.target.value.length === 0) {
+            sortedProps = unfilteredProps;
+        } else {
+            Object.keys(unfilteredProps).filter(x => {
+                if (x.toLowerCase().indexOf(event.target.value.toLowerCase()) > -1) {
+                    return true;
+                }
+                else {
+                    return false;
+                }
+            }).forEach(y => {
+                sortedProps[y] = unfilteredProps[y]
+            })
+        }
+        tdJSON.properties = sortedProps;
+        context.updateOfflineTD(JSON.stringify(tdJSON, null, 2))
```

Comment:
this line repeats so often that it almost deserves its own function

Reply:
will be done in next PR.